### PR TITLE
collect logs for Ramen Artifacts

### PR DESCRIPTION
**Pull request by yati1998**
This commits adds log collection of
Ramen artifacts for the hub cluster.
Here are the following details:
 * DRPolicy object
 * MirrorPeers object
 * Application namespace resources on the hub:
  - Channel(s) used by the Subscription(s)
  - All Subscription(s)
  - All PlacementRule(s)
  - DRPlacementControl
 * ACM resources across managed cluster namespaces as per drClusterSet in DRPolicy:
   - ManagedClusterView

BZ: #2017642

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Comment by yati1998** [comment]
@ShyamsundarR please let me know if it is fine or anything else needs to be added.

File: must-gather/collection-scripts/gather_clusterscoped_resources [hunk 1]
```diff
@@ -72,3 +72,23 @@ for command_desc in "${commands_desc[@]}"; do
...
+
+# Collect logs for drpolicy and mirrorpeers
+isMirrorPeer=$(timeout 60 oc get crd | grep mirrorpeers |awk '{print $1}')
+if [ "${isMirrorPeer}" == "mirrorpeers.multicluster.odf.openshift.io" ]; then
+        # Collect logs for oc get mirrorpeers
+        echo "collecting oc command get mirrorpeers" | tee -a "${BASE_COLLECTION_PATH}/gather-debug.log"
+        COMMAND_OUTPUT_FILE=${BASE_COLLECTION_PATH}/cluster-scoped-resources/oc_output/get_mirrorpeers
+        { oc get mirrorpeers; } >> "${COMMAND_OUTPUT_FILE}"
+        # Collect logs for oc get drpolicy
+        echo "collecting oc command get drpolicy" | tee -a "${BASE_COLLECTION_PATH}/gather-debug.log"
```
Comment (umangachapagain) [inline must-gather/collection-scripts/gather_clusterscoped_resources:84]: Existence of MirrorPeer doesn't guarantee existence of DRPolicy. Better to verify these separately.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: must-gather/collection-scripts/gather_clusterscoped_resources (hunk 1 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_clusterscoped_resources:84]: Ack!.

File: must-gather/collection-scripts/gather_clusterscoped_resources [hunk 2]
```diff
@@ -72,3 +72,23 @@ for command_desc in "${commands_desc[@]}"; do
      COMMAND_OUTPUT_FILE=${BASE_COLLECTION_PATH}/cluster-scoped-resources/oc_output/desc_${command_desc// /_}
     { oc describe "${command_desc}"; } >> "${COMMAND_OUTPUT_FILE}"
 done
+
+# Collect logs for drpolicy and mirrorpeers
```
Comment (ShyamsundarR) [inline must-gather/collection-scripts/gather_clusterscoped_resources:76]: (nit) comment should read collection of resources and not logs (in line 79 as well below)

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 3]
```diff
@@ -147,3 +147,25 @@ echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${B
 echo "collecting dump of oc get volumereplicationgroups all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplicationgroups --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vrg_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect vrg --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# Collect logs for Ramen DR Hub cluster Resources
+isMirrorPeer=$(timeout 60 oc get crd | grep mirrorpeers |awk '{print $1}')
+if [ "${isMirrorPeer}" == "mirrorpeers.multicluster.odf.openshift.io" ]; then
+            # For Channels of all namespaces
+            { oc get channel --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/channel_all_namespaces"
```
Comment (ShyamsundarR) [inline must-gather/collection-scripts/gather_namespaced_resources:155]: Technically collecting Channels, Subscriptions is useful, but I would assume this is an ACM must gather responsibility. We can stop at collecting DRPlacementControl and the PlacementRule.

I am not sure why we collect ManagedClusterAddOns here, am I missing something?

Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:155]: Not sure about the ACM must-gather, Collected it here as you mentioned this also needs to be collected. I don't know who is working on ACM must-gather, if there is anyone who can work on this in ACM must-gather, we can close this PR and rather use ACM must-gather. @ShyamsundarR please confirm this.

Comment (ShyamsundarR) [reply must-gather/collection-scripts/gather_namespaced_resources:155]: > Not sure about the ACM must-gather, Collected it here as you mentioned this also needs to be collected. I don't know who is working on ACM must-gather, if there is anyone who can work on this in ACM must-gather, we can close this PR and rather use ACM must-gather. @ShyamsundarR please confirm this.

We still need this, one option is to add all of this to the ACM must-gather for the hub. There is no one working on that, if we want to add these to the ACM hub must-gather **we** would need to add the same in there: https://github.com/stolostron/must-gather

Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:155]: okay sure. Then I would close this PR and will add it to https://github.com/stolostron/must-gather.

Comment (ShyamsundarR) [reply must-gather/collection-scripts/gather_namespaced_resources:155]: I would suggest we check if we can add it in there and then close this PR. Also, need to understand the release cycles for ACM must-gather such that we can be in sync.

**Comment by yati1998** [comment]
Closing this PR. Pushed another PR in ACM must-gather which is now merged as well.
@ShyamsundarR please do take a look at these PRs and let me know if anything else is required.
https://github.com/stolostron/must-gather/pull/86
https://github.com/stolostron/must-gather/pull/84

**Pull request closed without merging**